drivers/hdc1000/hdc1000.c

```diff
@@ -111,19 +119,42 @@ int hdc1000_get_results(const hdc1000_t *dev, int16_t *temp, int16_t *hum)
 
     if (status == HDC1000_OK) {
         /* if all ok, we convert the values to their physical representation */
+        uint16_t traw = ((uint16_t)buf[0] << 8) | buf[1];
+        temp_cached = (int16_t)((((int32_t)traw * 16500) >> 16) - 4000);
+
+        uint16_t hraw = ((uint16_t)buf[2] << 8) | buf[3];
+        hum_cached  = (int16_t)(((int32_t)hraw * 10000) >> 16);
+
         if (temp) {
-            uint16_t traw = ((uint16_t)buf[0] << 8) | buf[1];
-            *temp = (int16_t)((((int32_t)traw * 16500) >> 16) - 4000);
+            *temp = temp_cached;
         }
         if (hum) {
-            uint16_t hraw = ((uint16_t)buf[2] << 8) | buf[3];
-            *hum  = (int16_t)(((int32_t)hraw * 10000) >> 16);
+            *hum = hum_cached;
         }
     }
 
     return status;
 }
 
+int hdc1000_read_cached(const hdc1000_t *dev, int16_t *temp, int16_t *hum)
+{
+    /* check if initialization phase or outdated */
+    if ((last_read_time == 0xFFFFFFFF && hum_cached == 0 && temp_cached == 0) ||
+        (xtimer_now_usec() - last_read_time > HDC1000_RENEW_INTERVAL)) {
+        /* update last_read_time */
+        last_read_time = xtimer_now_usec();
+        return hdc1000_read(dev, temp, hum);
```

Comment:
Simply change this to
```c
return hdc1000_read(dev, &temp_cached, &hum_cached);
```
and remove all changes done to the `hdc100_get_results()` function.

Reply:
I think this is a great idea. Thanks!